multi-cluster-ingress/multi-cluster-blue-green-cluster/README.md

```diff
@@ -0,0 +1,355 @@
+
+This pattern could also be used cross-region, although that introduces concerns around data residency, so to keep things simple, this example demonstrates blue/green within a single region.
+
+### Relevant documentation
+
+- [Multi-cluster Ingress Concepts](https://cloud.google.com/kubernetes-engine/docs/concepts/ingress-for-anthos)
+- [Setting Up Multi-cluster Ingress](https://cloud.google.com/kubernetes-engine/docs/how-to/ingress-for-anthos-setup)
+- [Deploying Ingress Across Clusters](https://cloud.google.com/kubernetes-engine/docs/how-to/ingress-for-anthos)
+- [Google Cloud External HTTP(S) Load Balancing](https://cloud.google.com/load-balancing/docs/https)
+
+#### Versions
+
+- GKE clusters on GCP
+- All versions of GKE supported
+- Tested and validated with 1.18.10-gke.1500 on Nov 14th 2020
+
+### Networking Manifests
+
+This recipe demonstrates deploying Multi-cluster Ingress across two clusters to expose a single service hosted across both clusters. Unlike the [prior example](../multi-cluster-ingress-basic) both clusters (`gke-1` and `gke-3`) reside in the same GCP region (`us-west1`), although they're placed in two different zones. Both clusters sit behind the same MultiClusterIngress and load balancer IP, and the load balancer will round-robin traffic between both clusters, as they sit within the same GCP region.
+
+The two clusters in this example can be backends to MCI only if they are registered through Hub. Hub is a central registry of clusters that determines which clusters MCI can function across. A cluster must first be registered to Hub before it can be used with MCI.
+
+There are two Custom Resources (CRs) that control multi-cluster load balancing - the MultiClusterIngress (MCI) and the MultiClusterService (MCS). The MCI below describes the desired traffic matching and routing behavior.
+
+```yaml
+apiVersion: networking.gke.io/v1
+kind: MultiClusterIngress
+metadata:
+  name: foobar-ingress
+  namespace: multi-cluster-demo
+spec:
+  template:
+    spec:
+      backend:
+        serviceName: default-backend
+        servicePort: 8080
+```
+
+Similar to the Kubernetes Service, the MultiClusterService (MCS) describes label selectors and other backend parameters to group pods in the desired way. Unlike the [prior example](../multi-cluster-ingress-basic), in this recipe we're just going to use a single MCS, configured as the `default-backend`, to exhibit failover behavior. Notice the `clusters` annotation in the MCS definition below:
+
+```yaml
+apiVersion: networking.gke.io/v1
+kind: MultiClusterService
+metadata:
+  name: default-backend
+  namespace: multi-cluster-demo
+  annotations:
+    beta.cloud.google.com/backend-config: '{"default": "backend-health-check"}'
+spec:
+  template:
+    spec:
+      selector:
+        app: default-backend
+      ports:
+      - name: http
+        protocol: TCP
+        port: 8080
+        targetPort: 8080
+  clusters:
+  - link: "us-west1-a/gke-1"
+  - link: "us-west1-b/gke-3"
+```
```

Comment:
Since this `clusters:` section is the highlight of this recipe, we should also show a snippet for what changes we make to remove a cluster. For example, something like ... to remove a cluster from the backends of an MCS, we use the MCS's `clusters` field. When we apply the second version of the `foo-mcs` we remove the cluster that we want removed as a backend. Note that `us-west1-a/gke-1` no longer appears in this MCS:

``` 
apiVersion: networking.gke.io/v1
kind: MultiClusterService
metadata:
  name: default-backend
  namespace: multi-cluster-demo
  annotations:
    beta.cloud.google.com/backend-config: '{"default": "backend-health-check"}'
spec:
  template:
    spec:
      selector:
        app: default-backend
      ports:
      - name: http
        protocol: TCP
        port: 8080
        targetPort: 8080
  clusters:
  - link: "us-west1-b/gke-3"
```